src/Comms.cpp

```diff
@@ -1,15 +1,34 @@
 #include <Comms.h>
 
 namespace Comms {
+
+    std::map<int, commFunction> callbackMap;
+    EthernetUDP Udp;
+    char packetBuffer[sizeof(Packet)];
+
     void initComms() {
+        Ethernet.begin(mac, ip);
+        Udp.begin(port);
+    }
+
+    void registerCallback(int id, commFunction function) {
+        callbackMap.insert(std::pair<int, commFunction>(id, function));
     }
 
     uint32_t packetWaiting() {
-        return 0; // return 1 if packets waiting
+        return Udp.parsePacket();
     }
 
     void processPackets() {
-        // process all waiting packets
+        Udp.read(packetBuffer, sizeof(Packet));
+
+        Packet packet = *(struct Packet *)&packetBuffer;
+
+        //call callback function
+        uint16_t checksum = *(uint16_t *)&packet.checksum;
+        if (checksum == fletcher16(packet.data, packet.len)) {
```

Comment:
See my comment below too, but the checksum should include the id and the len of the packet. Might wanna just change the fletcher16 function to take in a packet pointer instead of a byte array.